patient-portal-backend/src/main/java/com/vaadin/demo/repositories/PatientRepository.java

```diff
@@ -2,6 +2,22 @@
 
 import com.vaadin.demo.entities.Patient;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
 
 public interface PatientRepository extends JpaRepository<Patient, Long> {
+    @Query("SELECT COUNT(p), '21-30' AS c FROM Patient p WHERE (p.birthDate >= '1987-01-01' AND p.birthDate <= '1997-01-01') UNION " +
```

Comment:
We can't rely on hard coded dates, this isn't right. 

The query needs to dynamically look up the age. Check AGE() and EXTRACT() postgres methods. Then do counts/group by that. Something along these lines. http://stackoverflow.com/questions/34651030/sql-query-to-group-by-age-range-from-date-created